core/unit_test/UnitTest_InitializeBackends.cpp

```diff
@@ -0,0 +1,108 @@
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <Kokkos_Core.hpp>
+
+#include <gtest/gtest.h>
+
+namespace {
+// The purpose of this test is to mimic Legion's use case of initializing and
+// finalizing individual backends
+TEST(initialization, separate_initialization) {
+  Kokkos::InitializationSettings kokkos_init_settings;
+  Kokkos::Impl::pre_initialize(kokkos_init_settings);
+#ifdef KOKKOS_ENABLE_SYCL
+  Kokkos::Experimental::SYCL::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  Kokkos::Experimental::OpenMPTarget::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_HIP
+  Kokkos::Experimental::HIP::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_CUDA
+  Kokkos::Cuda::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_HPX
+  Kokkos::Experimental::HPX::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_THREADS
+  Kokkos::Threads::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+  Kokkos::OpenMP::impl_initialize(kokkos_init_settings);
+#endif
+#ifdef KOKKOS_ENABLE_SERIAL
+  Kokkos::Serial::impl_initialize(kokkos_init_settings);
+#endif
+  Kokkos::Impl::post_initialize(kokkos_init_settings);
+
+#ifdef KOKKOS_ENABLE_SYCL
+  Kokkos::Experimental::SYCL::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  Kokkos::Experimental::OpenMPTarget::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_HIP
+  Kokkos::Experimental::HIP::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_CUDA
+  Kokkos::Cuda::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_HPX
+  Kokkos::Experimental::HPX::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_THREADS
+  Kokkos::Threads::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_OPENMP
+  Kokkos::OpenMP::impl_finalize();
+#endif
+#ifdef KOKKOS_ENABLE_SERIAL
+  Kokkos::Serial::impl_finalize();
+#endif
+
+  Kokkos::finalize();
```

Comment:
```suggestion
  Kokkos::finalize();
  
  EXPECT_FALSE(Kokkos::is_initialized());
  // ...
```

Reply:
Actually now that I look at it again I really dislike the semantics of `is_initialized`